rotkehlchen/api/rest.py

```diff
@@ -377,6 +377,18 @@ def inner(rest_api: 'RestAPI', **kwargs: Any) -> Response:
     return wrapper
 
 
+def register_post_download_cleanup(temp_file: str | Path) -> None:
```

Comment:
you should alway pass a `Path` here. The caller knows best if it's a string or a path. Most places should be a Path. This way you remove the if